### ISSUE TITLE
logger-objects: allow log functions

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#20436]
with config:
```toml
[tool.ruff.lint]
logger-objects = ['foo.get_it', 'AUTH']
```

The following use cases exists:
```python
from foo import get_it
get_it.info('foo %s' % (1,))   # detected

blah = get_it()
blah.info('foo %s' % (1,))  # not detected
get_it().info('foo %s' % (1,))  # not detected

blub = get_it
blub.info('foo %s' % (1,))  # also not detected

AUTH.info('foo %s' % (1,))  # also not detected
```

So, I would like logger-objects to be the result of a function/method call.
Furthermore I would like to specify the log variable names, our loggers are called `AUTH`, `NETWORK`, `ACL`, etc. and therefor don't match anything like `.*log.*`.
If I just put these names into logger-objects, they are ignored.

**Comment by ntBre** [comment]
I believe this is tracked by #12613.
